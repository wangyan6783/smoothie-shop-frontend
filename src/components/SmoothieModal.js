import React, { Component } from 'react';

class SmoothieModal extends Component {

    closeModal = (e) => {
        //event delegation
        if(e.target.matches("#smoothie-modal, #smoothie-close")) {
            this.props.updateSmoothie(null);
        }
    }

    render(){
        const { smoothie } = this.props;
        return (
            <div className="modal" id="smoothie-modal" onClick={this.closeModal}>
                <div className="modal-content-container">
                    <div className="modal-close"><span id="smoothie-close">x</span></div>
                    <div className="modal-content">
                        <div className="modal-slot-1">
                            <img src={require(`../assets/smoothies/${smoothie.imgUrl}-1.jpeg`)} alt={smoothie.name} className="modal-image"/>
                        </div>
                        <div className="modal-slot-2">
                            <div className="modal-title heading-2 mb-sm">{smoothie.name}</div>
                            <div className="modal-subtitle content-text-1 mb-sm">Tastes Like: {smoothie.tastes}</div>
                            <p className="modal-description content-text-2 mb-md">{smoothie.description}</p>
                            <button className="btn modal-button">Add to cart</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default SmoothieModal;