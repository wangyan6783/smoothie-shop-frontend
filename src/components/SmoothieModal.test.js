import React from "react";
import renderer from "react-test-renderer";
import SmoothieModal from "./SmoothieModal";

describe("SmoothieModal Component", () => {
  it("render correctly", () => {
    const smoothie = {
      name: "Tart Cherry + Raspberry",
      description:
        "This tangy smoothie hits hard with the tart bite of raspberries and sweet kick of our Mission Figs. Tart cherries and blueberries bring vibrant color to this antioxidant powerhouse, which helps put free-radicals in their place. Plus, the cherries help calm your system with anti-inflammatory properties and even help you sleep. Name another sorbet that can do that?",
      tastes: "Fresh Berry Sorbet",
      imgUrl: "tart-cherry-raspberry"
    };
    const smoothieModalComponent = renderer
      .create(<SmoothieModal smoothie={smoothie} />)
      .toJSON();
    expect(smoothieModalComponent).toMatchSnapshot();
  });
});
