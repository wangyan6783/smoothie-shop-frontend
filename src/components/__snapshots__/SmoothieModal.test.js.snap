// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmoothieModal Component render correctly 1`] = `
<div
  className="modal"
  id="smoothie-modal"
  onClick={[Function]}
>
  <div
    className="modal-content-container"
  >
    <div
      className="modal-close"
    >
      <span
        id="smoothie-close"
      >
        x
      </span>
    </div>
    <div
      className="modal-content"
    >
      <div
        className="modal-slot-1"
      >
        <img
          alt="Tart Cherry + Raspberry"
          className="modal-image"
          src="tart-cherry-raspberry-1.jpeg"
        />
      </div>
      <div
        className="modal-slot-2"
      >
        <div
          className="modal-title heading-2 mb-sm"
        >
          Tart Cherry + Raspberry
        </div>
        <div
          className="modal-subtitle content-text-1 mb-sm"
        >
          Tastes Like: 
          Fresh Berry Sorbet
        </div>
        <p
          className="modal-description content-text-2 mb-md"
        >
          This tangy smoothie hits hard with the tart bite of raspberries and sweet kick of our Mission Figs. Tart cherries and blueberries bring vibrant color to this antioxidant powerhouse, which helps put free-radicals in their place. Plus, the cherries help calm your system with anti-inflammatory properties and even help you sleep. Name another sorbet that can do that?
        </p>
        <button
          className="btn modal-button"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
